presto-main/src/test/java/io/prestosql/operator/TestCyclingGroupByHash.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import org.testng.annotations.Test;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestCyclingGroupByHash
+{
+    @Test
+    public void testSingleGroup()
+    {
+        CyclingGroupByHash groupByHash = new CyclingGroupByHash(1);
+        Page page = createPage(1);
+        GroupByIdBlock groupByIdBlock = computeGroupByIdBlock(groupByHash, page);
+        assertContentsEqual(groupByIdBlock, 0L);
+
+        page = createPage(2);
+        groupByIdBlock = computeGroupByIdBlock(groupByHash, page);
+        assertContentsEqual(groupByIdBlock, 0L, 0L);
+    }
+
+    @Test
+    public void testMultipleGroup()
+    {
+        CyclingGroupByHash groupByHash = new CyclingGroupByHash(2);
+        Page page = createPage(3);
+        GroupByIdBlock groupByIdBlock = computeGroupByIdBlock(groupByHash, page);
+        assertContentsEqual(groupByIdBlock, 0L, 1L, 0L);
+
+        page = createPage(2);
+        groupByIdBlock = computeGroupByIdBlock(groupByHash, page);
+        assertContentsEqual(groupByIdBlock, 1L, 0L);
+    }
+
+    private static void assertContentsEqual(GroupByIdBlock groupByIdBlock, long... groupIds)
```

Comment:
I think there already is a page equals somewhere, but it might be page equals page.  Maybe you could create a page from the longs and then use the existing page equals page... just an idea

Reply:
Yea, in PageAssertions we have assertPageEquals(). To use this we would have to create a page out of the GroupByIdBlock (since it is a block and not a page) as well as the expected group ids, just to do a page comparison. It feels a bit simpler here just to read out the groupIds as longs directly? But might be missing something.

Reply:
No need to change, but for future reference, I think you can do:
```
new GroupIdBlock(maxGroup, new IntArrayBlock(groupIds.length, Optional.empty(), groupIds));
```